src/main/java/bio/terra/service/dataset/DatasetTable.java

```diff
@@ -92,6 +94,38 @@ public DatasetTable primaryKey(List<Column> primaryKey) {
     return this;
   }
 
+  /**
+   * @param column with name as key
+   * @param row expected to contain a value associated with `column`
+   * @return value associated with `column`
+   */
+  private String columnValueFromRow(Column column, FieldValueList row) {
+    if (!row.hasSchema()) {
+      return "unknown";
+    }
+    try {
+      return row.get(column.getName()).getStringValue();
+    } catch (IllegalArgumentException e) {
+      return "unknown";
+    } catch (NullPointerException e) {
+      return "null";
+    }
+  }
+
+  /**
+   * @param row expected to contain values associated with the table's primary key
+   * @return a string representation of the table's primary key and its values
+   */
+  public String primaryKeyToString(FieldValueList row) {
+    if (primaryKey == null || primaryKey.isEmpty()) {
```

Comment:
You could also use `StringUtils.isEmpty` (from org.apache.commons.lang3) here which handles the null check

Reply:
Oh!  The `primaryKey` is not a string, it is a list of columns :)

Reply:
Oh of course! 🤦‍♀️